lib/verbalize/build.rb

```diff
@@ -0,0 +1,60 @@
+module Verbalize
+  class Build
+    class << self
+      def call(required_keywords = [], optional_keywords = [])
+        new(required_keywords, optional_keywords).call
+      end
+    end
+
+    def initialize(required_keywords, optional_keywords)
+      @required_keywords = required_keywords
+      @optional_keywords = optional_keywords
+    end
+
+    def call
+      <<-CODE
+def self.call(#{declaration_arguments_string})
+  __proxied_call(#{forwarding_arguments_string})
+end
+
+def self.call!(#{declaration_arguments_string})
+  __proxied_call!(#{forwarding_arguments_string})
+end
+
+def initialize(#{declaration_arguments_string})
+  #{initialize_body}
+end
+
+private
+
+attr_reader #{attribute_reader_string}
```

Comment:
might be better to pluralize this as `attribute_readers_string` to make it clear there can be more than one, but I don't feel strongly about it.